mptt/managers.py

```diff
@@ -619,48 +620,102 @@ def root_nodes(self):
         """
         return self._mptt_filter(parent=None)
 
-    @delegate_manager
-    def rebuild(self):
+    def _find_out_rebuild_fields(self):
         """
-        Rebuilds all trees in the database table using `parent` link.
+        Due to the behavior of the metaclass, it is not possible
+        to find out the fields in the __init__ correctly.
         """
+        lookups = self._translate_lookups(
+            left="left", right="right", level="level", tree_id="tree_id"
+        )
+        self._rebuild_fields = {value: key for key, value in lookups.items()}
+
+    def _get_parents(self, **filters):
         opts = self.model._mptt_meta
+        qs = self._mptt_filter(parent=None, **filters)
+        if opts.order_insertion_by:
+            qs = qs.order_by(*opts.order_insertion_by)
+        return list(qs.only("pk"))
 
-        qs = self._mptt_filter(parent=None)
+    def _get_children(self, **filters):
+        opts = self.model._mptt_meta
+        qs = self._mptt_filter(parent__isnull=False, **filters)
         if opts.order_insertion_by:
             qs = qs.order_by(*opts.order_insertion_by)
-        pks = qs.values_list("pk", flat=True)
 
-        rebuild_helper = self._rebuild_helper
-        idx = 0
-        for pk in pks:
-            idx += 1
-            rebuild_helper(pk, 1, idx)
+        children = defaultdict(list)
+        for child in qs.select_related("parent"):
```

Comment:
When I upgrade to version 0.15 i run into problem when the name of related is not "parent". Why not use opts.parent_attr?